src/templates/finops-hub/modules/dataFactory.bicep

```diff
@@ -179,147 +235,1364 @@ resource linkedService_storageAccount 'Microsoft.DataFactory/factories/linkedser
+// Export container extract pipeline + trigger
+// Trigger: New CSV files in exportContainer
+//
+// Queues the transform pipeline.
+// This pipeline must complete ASAP due to ADF's hard limit of 100 concurrent executions per pipeline.
+// If multiple large, partitioned exports run concurrently and this pipeline doesn't finish quickly, the transform pipeline won't get triggered.
+// Queuing up the transform pipeline and exiting immediately greatly reduces the likelihood of this happening.
+//------------------------------------------------------------------------------
+
+//------------------------------------------------------------------------------
+// Triggers
+//------------------------------------------------------------------------------
+resource trigger_exportContainer 'Microsoft.DataFactory/factories/triggers@2018-06-01' = {
+  name: extractExportTriggerName
+  parent: dataFactory
   dependsOn: [
-    linkedService_keyVault
+    stopHubTriggers
   ]
   properties: {
     annotations: []
+    pipelines: [
+      {
+        pipelineReference: {
+          referenceName: pipeline_extractExport.name
+          type: 'PipelineReference'
+        }
+        parameters: {
+          folderName: '@triggerBody().folderPath'
+          fileName: '@triggerBody().fileName'
+        }
+      }
+    ]
+    type: 'BlobEventsTrigger'
+    typeProperties: {
+      blobPathBeginsWith: '/${exportContainerName}/blobs/'
+      blobPathEndsWith: '.csv'
+      ignoreEmptyBlobs: true
+      scope: storageAccount.id
+      events: [
+        'Microsoft.Storage.BlobCreated'
+      ]
+    }
+  }
+}
+
+resource trigger_configContainer 'Microsoft.DataFactory/factories/triggers@2018-06-01' = {
+  name: updateConfigTriggerName
+  parent: dataFactory
+  dependsOn: [
+    stopHubTriggers
+  ]
+  properties: {
+    annotations: []
+    pipelines: [
+      {
+        pipelineReference: {
+          referenceName: pipeline_setup.name
+          type: 'PipelineReference'
+        }
+      }
+    ]
+    type: 'BlobEventsTrigger'
+    typeProperties: {
+      blobPathBeginsWith: '/${configContainerName}/blobs/'
+      blobPathEndsWith: 'settings.json'
+      ignoreEmptyBlobs: true
+      scope: storageAccount.id
+      events: [
+        'Microsoft.Storage.BlobCreated'
+      ]
+    }
+  }
+}
+
+resource trigger_daily 'Microsoft.DataFactory/factories/triggers@2018-06-01' = {
+  name: dailyTriggerName
+  parent: dataFactory
+  dependsOn: [
+    stopHubTriggers
+  ]
+  properties: {
+    pipelines: [
+      {
+        pipelineReference: {
+          referenceName: pipeline_get.name
+          type: 'PipelineReference'
+        }
+        parameters: {
+          Recurrence: 'Daily'
+        }
+      }
+    ]
+    type: 'ScheduleTrigger'
+    typeProperties: {
+      recurrence: {
+        frequency: 'Hour'
+        interval: 24
+        startTime: '2023-01-01T01:01:00'
+        timeZone: 'Pacific Standard Time'
```

Comment:
This is going to cause concerns from my team because it means everyone using hubs will pull data at the same time, which will look like a DDoS attempt. Do we have a way to naturally distribute it? Could we set the timezone to be based on the resource location? We may need PowerShell for that.

Reply:
Really good point.
I've added a module to determine the timezone based on the location.  It defaults to UTC for undefined regions.